kustomize-fns/virtual-service/function.go

```diff
@@ -0,0 +1,112 @@
+// Package virtual_service implements a kustomize function for configuring virtual services
+package virtual_service
+
+import (
+	"fmt"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"sigs.k8s.io/kustomize/kyaml/kio"
+
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+const (
+	Kind       = "VirtualServiceTransform"
+	APIVersion = "kubeflow.org/v1alpha1"
+	VSKind     = "VirtualService"
+)
+
+var _ kio.Filter = &VirtualServiceFunction{}
+
+// Filter returns a new VirtualServiceFunction
+func Filter() kio.Filter {
+	return &VirtualServiceFunction{}
+}
+
+// VirtualServiceFunction implements the ImagePrefix Function
+type VirtualServiceFunction struct {
```

Comment:
Would it make sense to use Istio client-go here instead of defining the structs? https://github.com/istio/client-g

Reply:
The structure isn't a virtual service or other defined ISTIO struct so I don't think we can use client go.